### PR TITLE
feat(e2e): Added  e2e config to disable mempool lanes

### LINKED ISSUE #4005: mempool/lanes: add lanes to e2e tests
The e2e application now takes the lane from the transaction's payload. All e2e tests have lanes enabled by default.

We need some e2e tests with an app that don't implement lanes, that is, it returns nil lane values on Info and CheckTx requests.

Tasks:
- [x] Extend manifest to support `no_lanes = true`, so that the app does not implement lanes.
- [x] Add the `no_lanes` option to the e2e generator, so that nightly tests run also with no lanes (in particular, combined with `abci_protocol = "tcp"`).
- [x] Add a `lanes=[..]` option to allow for custom lanes to be added to the manifest.

### PR DESCRIPTION
**Pull request by jmalicevic**
Closes #4005 .

This PR adds:
-  ~~one more test to confirm that the when lanes are disabled, the order of gossip and reaping is the same as in the old mempool version~~ (done in #4008)
- a config flag for the e2e manifests to disable lanes. That way we can run a network that has lanes disabled on top of this version of Comet and make sure it works and its performance is not degraded by it.

### CODE REVIEW

**Review by melekes — APPROVED** [review]
utACK

File: test/e2e/app/app.go [hunk 1]
```diff
@@ -178,11 +191,19 @@ func NewApplication(cfg *Config) (*Application, error) {
...
+		return &Application{
+			logger:    logger,
+			state:     state,
+			snapshots: snapshots,
+			cfg:       cfg,
+		}, nil
+	}
 
+	lanes, lanePriorities := LaneDefinitions(cfg.Lanes)
+	fmt.Println("APP Lane priorities", lanePriorities)
```
Comment (hvanz) [inline test/e2e/app/app.go:206]: remove debug message?

File: test/e2e/app/app.go [hunk 2]
```diff
@@ -323,7 +352,11 @@ func (app *Application) CheckTx(_ context.Context, req *abci.CheckTxRequest) (*a
 		time.Sleep(app.cfg.CheckTxDelay)
 	}
 
+	if app.cfg.DoNotUseLanes {
+		return &abci.CheckTxResponse{Code: kvstore.CodeTypeOK, GasWanted: 1}, nil
+	}
 	lane := extractLane(value)
+	fmt.Println("LANE ", lane)
```
Comment (hvanz) [inline test/e2e/app/app.go:359]: remove?

File: test/e2e/node/config.go [hunk 3]
```diff
@@ -50,6 +50,10 @@ type Config struct {
 
 	PbtsEnableHeight int64 `toml:"pbts_enable_height"`
 	PbtsUpdateHeight int64 `toml:"pbts_update_height"`
+
+	DoNotUseLanes bool `toml:"no_lanes"`
+
```
Comment (hvanz) [inline test/e2e/node/config.go:55]:
```suggestion
```

File: test/e2e/node/config.go [hunk 4]
```diff
@@ -82,6 +88,7 @@ func LoadConfig(file string) (*Config, error) {
 		Protocol:        "socket",
 		PersistInterval: 1,
 	}
+	fmt.Println(cfg)
```
Comment (hvanz) [inline test/e2e/node/config.go:91]: remove

Comment (hvanz) [inline test/e2e/node/config.go:91]:
```suggestion
```

File: test/e2e/app/app.go (hunk 2 above)
Comment (hvanz) [inline test/e2e/app/app.go:359]:
```suggestion
	fmt.Println("LANE ", lane)
```

File: mempool/iterators_test.go [hunk 5]
```diff
@@ -399,3 +399,46 @@ func TestReapOrderMatchesGossipOrder(t *testing.T) {
 	require.NoError(t, err)
 	require.Zero(t, mp.Size())
 }
+
+func TestReapOrderMatchesGossipOrderNolanes(t *testing.T) {
```
Comment (hvanz) [inline mempool/iterators_test.go:403]: Does this test belong to this PR?

Comment (hvanz) [reply mempool/iterators_test.go:403]: Ok, I thought the PR was only about e2e.

Comment (jmalicevic) [reply mempool/iterators_test.go:403]: I am creating a new PR for this ,it is cleaner

File: test/e2e/generator/generate.go [hunk 6]
```diff
@@ -318,6 +319,13 @@ func generateTestnet(r *rand.Rand, opt map[string]any, upgradeVersion string, pr
 		)
 	}
 
+	switch opt["no_lanes"] {
+	case false:
+		manifest.DoNotUseLanes = true
+	case true:
+		manifest.DoNotUseLanes = false
+	}
```
Comment (hvanz) [inline test/e2e/generator/generate.go:327]:
```suggestion
	manifest.DoNotUseLanes = !opt["no_lanes"]
```

**Review by hvanz — APPROVED** [review]
👍

**Comment by jmalicevic** [comment]
Blocked on #3638
